Codex follow-up review: thanks for pushing `4fc9e85`. The original non-string Codewars normalization path is improved now, and local merge/diff checks are clean, but I still can't approve workflows/deployment or merge yet.

I ran the targeted sync-service tests on the PR head:

`pytest tests/test_sync_service.py -q --maxfail=1`

The new test `tests/test_sync_service.py::test_sync_codewars_handles_null_value` fails. With `{"codewars": None}`, the code normalizes the username to an empty string, does not enqueue a Codewars fetch job, and marks Codewars as `skipped`; the test currently expects `failed`:

`AssertionError: assert 'skipped' == 'failed'`

Please align the expected behavior/test with the service behavior. If a null/cleared username should be treated as an explicit failed sync, the service needs to mark it that way; otherwise the test should expect `skipped`. I am leaving the waiting GitHub workflows and Vercel authorization untouched until this is green.

--- a/tests/test_sync_service.py
+++ b/tests/test_sync_service.py
@@ -443,16 +443,13 @@ def test_sync_codewars_handles_non_string_value(monkeypatch):
 
 def test_sync_codewars_handles_null_value(monkeypatch):
     """A null/None codewars value must be normalized to empty string without
-    raising AttributeError."""
+    raising AttributeError. An empty username means no fetch job is created,
+    matching the same skipped behavior as other platforms."""
     now = datetime.now(timezone.utc)
     user = FakeUser()
     db = FakeDB()
     cache = FakeCache()
 
-    monkeypatch.setattr(
-        "app.profile.sync_service.fetch_codewars",
-        lambda username: None,
-    )
     monkeypatch.setattr("app.profile.sync_service.invalidate_leaderboard_cache", lambda: None)
 
     payload, status_code = sync_user_platforms(
@@ -464,5 +461,5 @@ def test_sync_codewars_handles_null_value(monkeypatch):
     )
 
     assert status_code == 200
-    # None results in empty string username → fetch returns None → no data → failed
-    assert payload["platforms"]["codewars"]["status"] == "failed"
+    # None normalizes to empty string → no fetch job enqueued → skipped
+    assert payload["platforms"]["codewars"]["status"] == "skipped"

